From 50ed977e1b166ba23a029f3a5194121ea4d45dd7 Mon Sep 17 00:00:00 2001
From: Thomas Perl <m@thp.io>
Date: Fri, 6 Feb 2015 16:28:43 +0100
Subject: [PATCH] zypp/PublicKey.cc: Use GPG_BINARY from KeyRing.cc

On systems where "gpg" is not available, checking the public key
would fail. Use GPG_BINARY in both PublicKey.cc and KeyRing.cc,
and define it in a single place only (ExternalProgram.h).
---
 zypp/ExternalProgram.h | 2 ++
 zypp/KeyRing.cc        | 2 --
 zypp/PublicKey.cc      | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/zypp/ExternalProgram.h b/zypp/ExternalProgram.h
index bc62a4c..46e8145 100644
--- a/zypp/ExternalProgram.h
+++ b/zypp/ExternalProgram.h
@@ -22,6 +22,8 @@
 #include "zypp/base/ExternalDataSource.h"
 #include "zypp/Pathname.h"
 
+#define GPG_BINARY "/usr/bin/gpg2"
+
 namespace zypp {
 
     /**
diff --git a/zypp/KeyRing.cc b/zypp/KeyRing.cc
index 13cae3d..a4dc447 100644
--- a/zypp/KeyRing.cc
+++ b/zypp/KeyRing.cc
@@ -37,8 +37,6 @@ using std::endl;
 #undef  ZYPP_BASE_LOGGER_LOGGROUP
 #define ZYPP_BASE_LOGGER_LOGGROUP "zypp::KeyRing"
 
-#define GPG_BINARY "/usr/bin/gpg2"
-
 ///////////////////////////////////////////////////////////////////
 namespace zypp
 { /////////////////////////////////////////////////////////////////
diff --git a/zypp/PublicKey.cc b/zypp/PublicKey.cc
index 4d3d773..52d9d7a 100644
--- a/zypp/PublicKey.cc
+++ b/zypp/PublicKey.cc
@@ -349,7 +349,7 @@ namespace zypp
         static filesystem::TmpDir dir;
         const char* argv[] =
         {
-          "gpg",
+          GPG_BINARY,
           "-v",
           "--no-default-keyring",
           "--fixed-list-mode",
-- 
2.1.4

